lib/job_helpers/email_job_helper.rb

```diff
@@ -0,0 +1,19 @@
+module EmailJobHelper
+  private
+
+  def send_email(res)
+    UserMailer.reservation_status_update(res).deliver_now
+  end
+
+  # queries can just be a list of symbols like :checked_out, :not_overdue
+  # but if you want to send arguments to the query method, wrap them in an array
+  # like [:not_flagged, :missed_email_sent]
+  def collection(*queries) 
+    @set ||= queries.inject(Reservation) { |result, query| result.send(*query) }
+  end
+
+  def clean
+    @set = nil
+  end
```

Comment:
Nice, so you'd be including this in each of the email jobs? How would you pass a custom `where` finder, something like:

```
collection([:checked_out, [:where, 'due_date > ?', Time.zone.today]])
```

Overall the methods are nice and compact, why don't you put it all together and we'll see what it looks like?